src/status_im2/contexts/add_new_contact/events_test.cljs

```diff
@@ -10,47 +13,160 @@
-              :id       ens
-              :type     :ens-name
-              :ens-name ens-stateofus-eth}))
-
-(deftest search-empty-string-test
-  (is (= (core/set-new-identity {:db {:contacts/new-identity :foo}} "")
-         {:db {}})))
-
-(deftest search-uncompressed-key-test
-  (is (= (core/set-new-identity {:db {}} ukey)
-         {:db {:contacts/new-identity
-               {:input      ukey
-                :public-key ukey
-                :state      :error
-                :error      :uncompressed-key}}})))
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; unit tests (validate-contact fn)
+
+(deftest validate-contact-test
+  (are [i e] (= (core/validate-contact (core/init-contact
+                                        {:user-public-key user-ukey :input i}))
+                (core/init-contact e))
+
+   ""            {:user-public-key user-ukey
+                  :input           ""
+                  :type            :empty
+                  :state           :empty}
+
+   " "           {:user-public-key user-ukey
+                  :input           " "
+                  :type            :empty
+                  :state           :empty}
+
+   ukey          {:user-public-key user-ukey
+                  :input           ukey
+                  :id              ukey
+                  :type            :public-key
+                  :public-key      ukey
+                  :state           :invalid
+                  :msg             :t/not-a-chatkey}
+
+   ens           {:user-public-key user-ukey
+                  :input           ens
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}
+
+   (str " " ens) {:user-public-key user-ukey
+                  :input           (str " " ens)
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}
+
+   ckey          {:user-public-key user-ukey
+                  :input           ckey
+                  :id              ckey
+                  :type            :compressed-key
+                  :state           :decompress-key}
+
+   link-ckey     {:user-public-key user-ukey
+                  :input           link-ckey
+                  :id              ckey
+                  :type            :compressed-key
+                  :state           :decompress-key}
+
+   link-ens      {:user-public-key user-ukey
+                  :input           link-ens
+                  :id              ens
+                  :type            :ens
+                  :ens             ens-stateofus-eth
+                  :state           :resolve-ens}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; event handler tests (set-new-identity, set-new-identity-success/error)
```

Comment:
@erikseppanen, to follow the Clojure Style Guide and the Lisp tradition, please use section comments (with 3 or 4 semicolons), like so:

```clojure
;;;; Event handler tests
```

Maybe you already know this, but as an Emacs user, you can enable the built-in `outline-minor-mode` in Clojure buffers and if you add `(setq outline-minor-mode-highlight 'override)` to your config, you'll see section comments with a different color. With the minor mode enabled you'll also get collapsible sections, etc.

Reply:
@ilmotta Yes, I usually do this, and if I end up keeping the comments I'll update them accordingly.

I'll try outline-minor-mode, I think I'll really really like that.

Reply:
Converted to 3 semicolon sections. Thanks for the tip on `outline-minor-mode`!